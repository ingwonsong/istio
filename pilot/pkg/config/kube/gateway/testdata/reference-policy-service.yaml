apiVersion: gateway.networking.k8s.io/v1alpha2
kind: Gateway
metadata:
  name: gateway
  namespace: istio-system
spec:
  addresses:
  - value: istio-ingressgateway
    type: Hostname
  gatewayClassName: istio
  listeners:
  - name: simple
    hostname: "simple.domain.example"
    port: 80
    protocol: HTTP
    allowedRoutes:
      namespaces:
        from: All
---
apiVersion: gateway.networking.k8s.io/v1alpha2
kind: ReferencePolicy
metadata:
  name: allow-service
  namespace: service
spec:
  from:
  - group: gateway.networking.k8s.io
    kind: HTTPRoute
    namespace: istio-system
  to:
  - group: ""
    kind: Service
    name: my-svc
---
apiVersion: gateway.networking.k8s.io/v1alpha2
kind: HTTPRoute
metadata:
  name: http
  namespace: istio-system
spec:
  parentRefs:
  - name: gateway
    namespace: istio-system
  hostnames: ["simple.domain.example"]
  rules:
  - backendRefs:
    - name: my-svc
      namespace: service
      port: 80
