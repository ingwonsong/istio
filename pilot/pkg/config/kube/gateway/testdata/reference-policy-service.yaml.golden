apiVersion: networking.istio.io/v1alpha3
kind: Gateway
metadata:
  annotations:
    internal.istio.io/gateway-service: istio-ingressgateway.istio-system.svc.domain.suffix
    internal.istio.io/parent: Gateway/gateway/simple.istio-system
  creationTimestamp: null
  name: gateway-istio-autogenerated-k8s-gateway-simple
  namespace: istio-system
spec:
  servers:
  - hosts:
    - '*/simple.domain.example'
    port:
      name: default
      number: 80
      protocol: HTTP
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  annotations:
    internal.istio.io/parent: HTTPRoute/http.istio-system
    internal.istio.io/route-semantics: gateway
  creationTimestamp: null
  name: http-0-istio-autogenerated-k8s-gateway
  namespace: istio-system
spec:
  gateways:
  - istio-system/gateway-istio-autogenerated-k8s-gateway-simple
  hosts:
  - simple.domain.example
  http:
  - route:
    - destination:
        host: my-svc.service.svc.domain.suffix
        port:
          number: 80
---
